akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -77,32 +120,107 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with GenericOut
       case x ⇒ receivedUnexpectedFrame(x)
     }
   }
-  sealed abstract class ReceivingData(outlet: BufferedOutlet[ByteString], afterEndStreamReceived: IncomingStreamState) extends IncomingStreamState { _: Product ⇒
+  class Buffer(streamId: Int, outlet: SubSourceOutlet[ByteString]) extends OutHandler {
```

Comment:
just naming of the class, "Buffer" is a bit bland, though I guess it's only used here hm... still :)